apps/designer/app/designer/features/style-panel/controls/font-weight/font-weight-control.tsx

```diff
@@ -0,0 +1,103 @@
+import { Select } from "@webstudio-is/design-system";
+import { FontWeight, fontWeights } from "@webstudio-is/fonts";
+import { toValue } from "@webstudio-is/react-sdk";
+import { useMemo } from "react";
+import { useAssets } from "~/designer/shared/assets";
+import { getFinalValue } from "../../shared/get-final-value";
+import type { ControlProps } from "../../style-sections";
+
+const allFontWeights = (Object.keys(fontWeights) as Array<FontWeight>).map(
+  (weight) => ({
+    label: `${fontWeights[weight].label} (${weight})`,
+    weight,
+  })
+);
+
+const useFontWeights = (currentFamily: string, currentWeight: string) => {
+  const { assets } = useAssets("font");
+
+  // Find all font weights that are available for the current font family.
+  const availableFontWeights = useMemo(() => {
+    const found = allFontWeights.filter((option) => {
+      return assets.find((asset) => {
+        return (
+          "meta" in asset &&
+          "family" in asset.meta &&
+          asset.meta.family === currentFamily &&
+          String(asset.meta.weight) === option.weight
+        );
+      });
+    });
+    return found.length === 0 ? allFontWeights : found;
+  }, [currentFamily, assets]);
+
+  const labels = useMemo(
+    () => availableFontWeights.map((option) => option.label),
+    [availableFontWeights]
+  );
+
+  const selectedLabel = useMemo(() => {
+    const selectedOption = availableFontWeights.find(
+      (option) => option.weight == currentWeight
+    );
+    return selectedOption?.label;
+  }, [currentWeight, availableFontWeights]);
+
+  return { labels, selectedLabel, availableFontWeights };
+};
+
+export const FontWeightControl = ({
+  currentStyle,
+  inheritedStyle,
+  setProperty,
+  styleConfig,
+}: ControlProps) => {
+  // @todo show which instance we inherited the value from
+  const fontWeight = getFinalValue({
+    currentStyle,
+    inheritedStyle,
+    property: styleConfig.property,
+  });
+
+  // We need the font family to determine which font weights are available
+  const fontFamily = getFinalValue({
+    currentStyle,
+    inheritedStyle,
+    property: "fontFamily",
+  });
+
+  const { labels, selectedLabel, availableFontWeights } = useFontWeights(
+    toValue(fontFamily, { withFallback: false }),
+    toValue(fontWeight)
+  );
+
+  if (fontWeight === undefined) return null;
```

Comment:
Is this just for TS, or is it possible for current `fontWeight` to be `undefined`? If possible, is it the right behaviour to not render the control in such case?

Reply:
I believe its possible, depending on what getFinalValue() does now or will do in the future, right now it uses both, style from our data and from the computed style browser, so if non of them have that value, it can be undefined

Reply:
oh yeah, we need to think later about this check in all controls, atm it never happened, maybe because there is always a value, so in this case its just typescript satisfaction:
e.g. getComputedStyle(document.body.parentElement) returns more properties than we support in style panel, things that aren't set have initial value setup, so I think it actually returns the entire CSS, even though the spec says it only returns things that apply. Maybe it doesn't return properties that can't be applied to a certain type of element at all ..
its an edge case in any case, so maybe we should just silence typescript and remove this check everywhere

Reply:
should be separate pr though, because this check is in all controls